### PR TITLE
Shell cleanup/fixes.

### PR DESCRIPTION
**Pull request by thz**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: api/hack/ci/ci-push-images.sh [hunk 1]
```diff
@@ -1,47 +1,39 @@
...
+#!/bin/sh
 
 set -euo pipefail
-export GIT_HEAD_HASH="$(git rev-parse HEAD|tr -d '\n')"
-cd $(dirname $0)/../../..
+GIT_HEAD_HASH="$(git rev-parse HEAD)"
+here=$(dirname "$0")
+cd "$here/../../.."
 
-source ./api/hack/lib.sh
```
Comment (alvaroaleman) [inline api/hack/ci/ci-push-images.sh:7]: Whats the advantage of `.` over `source`?

File: api/hack/ci/ci-push-images.sh [hunk 2]
```diff
@@ -1,47 +1,39 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
 set -euo pipefail
-export GIT_HEAD_HASH="$(git rev-parse HEAD|tr -d '\n')"
-cd $(dirname $0)/../../..
+GIT_HEAD_HASH="$(git rev-parse HEAD)"
+here=$(dirname "$0")
+cd "$here/../../.."
```
Comment (alvaroaleman) [inline api/hack/ci/ci-push-images.sh]: That is confusing, `here` is not `here` anymore if you go to `there`

File: api/hack/ci/ci-push-images.sh (hunk 1 above)
Comment (thz) [reply api/hack/ci/ci-push-images.sh:7]: It is POSIX compliant and satisfies shellcheck.

File: api/hack/ci/ci-push-images.sh (hunk 2 above)
Comment (thz) [reply api/hack/ci/ci-push-images.sh]: `here` as in where the shell-script is.

Comment (alvaroaleman) [reply api/hack/ci/ci-push-images.sh]: what about `my-dir`? `here` is describing something in relative terms, but you are describing something that is absolute

Comment (thz) [reply api/hack/ci/ci-push-images.sh]: ~We can spare the whole discussion anyways because a simple `cd ../../..` does the trick, right?~ ;-)

Comment (thz) [reply api/hack/ci/ci-push-images.sh]: Aehm. Ignore my last comment. ~I'll go for $base/ then...~

Comment (thz) [reply api/hack/ci/ci-push-images.sh]: How would you like using `git rev-parse --show-toplevel` instead of some `../../..`?

Comment (alvaroaleman) [reply api/hack/ci/ci-push-images.sh]: That sounds awesome

Comment (thz) [reply api/hack/ci/ci-push-images.sh]: 1d7d6e8297255f660c287173c577c0eb2741c57f

**Pull request merged**